src/ml_board/frontend/Dashboard/src/worker_socket/MainThreadCallbacks.ts

```diff
@@ -14,53 +14,65 @@ const initialStateForGraphs: evalResultCustomData = {
     colors_mapped_to_exp_id: {}
 };
 
+// Hashing is faster instead of switching over the the eventType
+const MapEventToProcess = {
+    [MLGYM_EVENT.JOB_STATUS]: (dataToRedux: DataToRedux, data: any) => { dataToRedux.jobStatusData = handleJobStatusData(data) },
```

Comment:
not entirely sure how typescript defines maps, but wouldn't  this be list key? `[MLGYM_EVENT.JOB_STATUS]`

Reply:
This is just the way to tell the transpiler to evaluate _**MLGYM_EVENT.JOB_STATUS**_ instead of taking it as a literal key.

So for example here, the key now is "job_status" and not "MLGYM_EVENT.JOB_STATUS"

Lastly, to be clear with terminology I'm using a normal Object not a Map because after checking out the differences, I found the Map doesn't provide more value to use in this specific case. (we just want the indexing O(1) and Objects do that as far as I looked).

Reply:
ok thanks for the clarification :-)